### PR TITLE
CHEF-65: Create inspec-5 release branch in Expeditor and Dependabot configuration

### PR DESCRIPTION
**Pull request by Vasu1105**
## Description

This is to update the expeditor and dependabot configuration to get ready for InSpec 6 release.

It creates an inspec-5 release branch in the CI configuration. Cutting a branch is a simple operation after that.

This is the first phase in preparation for merging inspec-6 to main.
Roughly:
1. Branch inspec-5 from main; temporarily freeze main
2. Verify CI health of inspec-5 pipelines
3. Merge inspec-6 to main
4. Verify CI health of main pipelines
5. main thaws; development targets InSpec 6 and some things backport to inspec-5 and inspec-4 pending the release of InSpec 6.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: .expeditor/config.yml [hunk 1]
```diff
@@ -78,9 +78,9 @@ github:
    - "Expeditor: Bump Major Version"
 
 release_branches:
-  - inspec-6:
```
Comment (clintoncwolfe) [inline .expeditor/config.yml:81]: At this time, we should not delete the inspec-6 release branch from the configuration. We want to leave it untouched, matching version 6.* .

File: .expeditor/config.yml [hunk 2]
```diff
@@ -78,9 +78,9 @@ github:
...
-     version_constraint: 6.*
   - main:
+     version_constraint: 6.*
```
Comment (clintoncwolfe) [inline .expeditor/config.yml:84]: We want to change the version constrain on main to be a dummy value, `99.*` or similar. We do not want to main to match anything at the moment. Main will be temporarily unreleasable until we merge a PR from the inspec-6 branch making main match 6.*

File: .github/dependabot.yml [hunk 3]
```diff
@@ -25,13 +25,13 @@ updates:
     interval: daily
   open-pull-requests-limit: 10
 - package-ecosystem: bundler
-  target-branch: "inspec-6"
+  target-branch: "inspec-5"
```
Comment (clintoncwolfe) [inline .github/dependabot.yml:28]: Very good catch to configure Dependabot as well.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -25,13 +25,13 @@ updates:
     interval: daily
   open-pull-requests-limit: 10
 - package-ecosystem: bundler
-  target-branch: "inspec-6"
```
Comment (clintoncwolfe) [inline .github/dependabot.yml:28]: At this time we are not deleting the InSpec 6 branch - that will happen in a merge.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]

**Comment by Vasu1105** [comment]
Thanks @clintoncwolfe !! Updated the PR as per review comments.

**Review by clintoncwolfe — APPROVED** [review]
Looks good - thank you. Ship it!

**Pull request merged**